camel/agents/chat_agent.py

```diff
@@ -27,6 +30,22 @@
 )
 
 
+@dataclass(frozen=True)
+class ChatAgentResponse:
```

Comment:
I have some questions:
- Why do not we name it `MessagesWithExtras`? 
- Should we put it into the messages subpackage?
- Can we inherit from `tuple` to avoid breaking and provide a better way to unpack?

Reply:
> Why do not we name it MessagesWithExtras?

The name is self-descriptive IMO. What would you propose?

> Should we put it into the messages subpackage?

I think no. I prefer the inferred classes to reside along with the inferring class.

> Can we inherit from tuple to avoid breaking and provide a better way to unpack? (maybe trick since tuple is immutable)

Dataclasses is the best way to organize struct records that I know.